programs/client/FuzzLoop.cpp

```diff
@@ -536,22 +538,27 @@ bool Client::buzzHouse()
                 std::uniform_int_distribution<uint32_t> next_dist(1, prob_space);
                 const uint32_t nopt = next_dist(rg.generator);
 
+                if ((correctness_oracle + settings_oracle + dump_oracle + peer_oracle) > 0
+                    && nopt < (correctness_oracle + settings_oracle + dump_oracle + peer_oracle + 1))
+                {
+                    (void)qo.ResetOracleValues();
```

Comment:
The fact that you need to do all these nasty casts means none of those methods should return an int at all